### ISSUE TITLE
Simple Collection Literals

### ISSUE THREAD

**Issue by HughG** [Kotlin/KEEP#230]
Issue for tracking and discussing my pull request for a simple mechanism for collection literals, #229.

**Comment by gildor** [comment]
Should it be discussed as part of [an existing proposal for collection literals](https://github.com/Kotlin/KEEP/issues/113)?

Having 2 proposals for the same thing doesn't look like a good idea, it would be very hard to follow the discussion

**Comment by HughG** [comment]
@gildor, I did consider doing that, having read the existing proposal in detail.  However, my proposal is very different from the existing one except in terms of the surface syntax and the overall goal of "have collection literals".  I couldn't think of an obvious way to "merge" my proposal with Ben Leggiero's, and the issues and open questions I have for my proposal are very different from those for Ben's.  Therefore I decided that it would be more confusing to have a single discussion thread about the details of two different proposals.

However, I could be persuaded and, if someone wanted to merge the proposals somehow, it might make sense to merge the discussions.  In that case I would take the effort to summarise the main points from one discussion as a new comment on the other proposal discussion thread.

If people want to discuss the overall topic of collection literals, or other approaches, I humbly suggest it probably makes more sense to do that on the Kotlin Slack or Discussion forums.  But of course the lines aren't clear.

**Comment by gildor** [comment]
I think none of those proposals would achieve nothing without merge (or just closing one of them)

**Comment by elizarov** [comment]
I'm closing this issue and the corresponding PR in favor of moving the discussion from the subject of a particular design to the discussion of use-cases and requirements that such a design shall meet. So far, the proposed design does not meet all the basic requirements that were identified for support of collection literals in Kotlin.

Please, follow the YouTrack issue https://youtrack.jetbrains.com/issue/KT-43871 for further details and cast your votes there 👍
